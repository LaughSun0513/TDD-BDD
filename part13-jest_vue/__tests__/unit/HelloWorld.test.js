import { shallowMount } from "@vue/test-utils";
import HelloWorld from "@/components/HelloWorld";
import Vue from "vue";

describe("HelloWorld.vue", () => {
	// it("DOM测试只能检测UI层面的变化 不方便", () => {
	// 	const root = document.createElement("div");
	// 	root.className = "root";
	// 	document.body.appendChild(root);
	// 	new Vue({
	// 		render: h => (
	// 			HelloWorld, {
	// 				props: {
	// 					msg: "Yux"
	// 				}
	// 			}
	// 		)
	// 	}).$mount(".root");
	// 	console.log(document.body.innerHTML);
	// 	expect(document.getElementsByClassName('hello').length).toBe(1);
	// });

	it("利用shallowMount浅渲染HelloWorld组件,不渲染HelloWorld子组件,适合单元测试", () => {
		const msg = "hahahah";
		// shallowMount浅渲染组件
		const wrapper = shallowMount(HelloWorld, {
			propsData: {
				msg
			}
		});

		// wrapper上的方法 props/findAll/setProps
		expect(wrapper.props('msg')).toEqual(msg);
		expect(wrapper.findAll('.myTitle').length).toBe(1);

		wrapper.setProps({
			msg: 'Yux'
		});
		expect(wrapper.props('msg')).toEqual('Yux');
	});

	it("复习:利用toMatchSnapshot快照进行组件的渲染", () => {
		const wrapper = shallowMount(HelloWorld, {
			propsData: {
				msg: 'Yux-snapshot'
			}
		});
		expect(wrapper).toMatchSnapshot();
	});
});