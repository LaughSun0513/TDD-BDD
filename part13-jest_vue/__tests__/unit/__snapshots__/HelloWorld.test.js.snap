// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue 复习:利用toMatchSnapshot快照进行组件的渲染 1`] = `
<div class="hello">
  <h1 class="myTitle">Yux-snapshot</h1>
</div>
`;
